### PR TITLE
Hollow Knight: update item name groups

### PR DESCRIPTION
**Pull request by kindasneaki**
## What is this fixing or adding?
Adding in missing item name groups and fixing the names of the ones that were currently there.

## How was this tested?
ran webhost and looked at the datapackage

### CODE REVIEW

**Comment by kindasneaki** [comment]
This is a draft at the moment because I dont know if the names are what is wanted

File: worlds/hk/Items.py [hunk 1]
```diff
@@ -19,18 +19,38 @@ class HKItemData(NamedTuple):
...
+    "Charms": lookup_type_to_names["Charm"],
+    "CharmNotches": lookup_type_to_names["Notch"],
+    "Claw": {x + "Mantis_Claw" for x in directionals},
+    "Cloak": {x + "Mothwing_Cloak" for x in directionals} | {"Shade_Cloak", "Split_Shade_Cloak"},
+    "LifebloodCocoons": lookup_type_to_names["Cocoon"],
     "Dreamers": {"Herrah", "Monomon", "Lurien"},
-    "Cloak": {x + 'Mothwing_Cloak' for x in directionals} | {'Shade_Cloak', 'Split_Shade_Cloak'},
-    "Claw": {x + 'Mantis_Claw' for x in directionals},
-    "CDash": {x + 'Crystal_Heart' for x in directionals},
     "Fragments": {"Queen_Fragment", "King_Fragment", "Void_Heart"},
```
Comment (BadMagic100) [inline worlds/hk/Items.py:33]: Would probably be better for this to be WhiteFragments to distinguish from vessel fragments, while we're changing things

File: worlds/hk/Items.py [hunk 2]
```diff
@@ -19,18 +19,38 @@ class HKItemData(NamedTuple):
...
+    "Grubs": lookup_type_to_names["Grub"],
+    "JournalEntries": lookup_type_to_names["Journal"],
+    "JunkPitChests": lookup_type_to_names["JunkPitChest"],
+    "Keys": lookup_type_to_names["Key"],
+    "LoreTablets": lookup_type_to_names["Lore"],
+    "MaskShards": lookup_type_to_names["Mask"],
+    "Maps": lookup_type_to_names["Map"],
+    "Mimics": lookup_type_to_names["Mimic"],
+    "Nail": lookup_type_to_names["CursedNail"],
+    "PalaceLore": lookup_type_to_names["PalaceLore"],
```
Comment (BadMagic100) [inline worlds/hk/Items.py:46]: Could palace totems and journal entry have this kind of handling as well or do the lookup types not have it?

Also I'm not opposed to having this group but I think that LoreTablets should include palace tablets as well.

File: worlds/hk/Items.py
Comment (BadMagic100) [inline worlds/hk/Items.py:1]: Would probably be nice to have groups for fireball, dive, and scream in line with https://github.com/ArchipelagoMW-HollowKnight/Archipelago.HollowKnight/issues/74

**Review by BadMagic100 — CHANGES_REQUESTED** [review]
Names broadly look fine to me, just a couple other changes I'd make while we're messing with the groups anyway

File: worlds/hk/Items.py (hunk 2 above)
Comment (kindasneaki) [reply worlds/hk/Items.py:46]: I take it you mean something like this?
```             
                "PalaceJournal": [
                    "Journal_Entry-Seal_of_Binding"
                ],
                "PalaceLore": [
                    "Lore_Tablet-Palace_Throne",
                    "Lore_Tablet-Palace_Workshop",
                    "Lore_Tablet-Path_of_Pain_Entrance"
                ],
                "PalaceTotem": [
                    "Soul_Totem-Palace"
                ],
```
Or should it just be a group with just Palace? I'm not sure what the best name to call it for this though

Comment (BadMagic100) [reply worlds/hk/Items.py:46]: I think this would be fine, yeah. Though I think probably PoP totems need to be in the palacetotems group too

**Comment by BadMagic100** [comment]
This would resolve #2214 by the way, let's see if GitHub will link it lol

File: worlds/hk/Items.py (hunk 2 above)
Comment (kindasneaki) [reply worlds/hk/Items.py:48]: I've never done path of pain so good catch.

File: worlds/hk/Items.py
Comment (kindasneaki) [reply worlds/hk/Items.py:1]: Added these three in, let me know if the names work

**Review by kindasneaki — COMMENTED** [review]
Hopefully I got everything this time so I can open the PR

**Comment by BadMagic100** [comment]
Yep looks good to me

**Review by BadMagic100 — APPROVED** [review]

**Review by black-sliver — APPROVED** [review]
approved by BadMagic100 and i don't see anything wrong with it

**Pull request merged**